### PR TITLE
GH-3763: Add `handleReactive()` for Java DSL

### LINKED ISSUE #3763: Feature Request - handleReactive(ReactiveMessageHandler)
**Expected Behavior**
I'd like there to be a reactive `handleReactive(ReactiveMessageHandler)` operator that will enable me to easily write reactive code as a handler.

**Current Behavior**
There is a need to use the `ReactiveMessageHandlerAdapter` class and write the code inside of it.

**Context**
The workaround is pretty simple, but an implementation of such a function seems to be pretty simple either. Here's a [SO question to support the issue](https://stackoverflow.com/questions/71649191/reactive-spring-integration-access-flux-in-the-middle-of-an-integrationflow).

### PR DESCRIPTION
**Pull request by artembilan**
Fixes https://github.com/spring-projects/spring-integration/issues/3763

Add a convenient terminal operator to `BaseIntegrationFlowDefinition` based on a `ReactiveMessageHandler`.
Also add an overload like `handleReactive(ReactiveMessageHandlerSpec)` to let end-user to choose a protocol-specific channel adapter

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/dsl/BaseIntegrationFlowDefinition.java [hunk 1]
```diff
@@ -2916,6 +2920,68 @@ public IntegrationFlow nullChannel() {
 				.get();
 	}
 
+	/**
+	 * Populate a terminal consumer endpoint for the selected protocol specific
+	 * {@link MessageHandler} implementation
+	 * from the respective namespace factory (e.g. {@code Http, Kafka, Files}).
+	 * In addition, accept options for the integration endpoint using {@link GenericEndpointSpec}.
+	 * @param messageHandlerSpec the {@link MessageHandlerSpec} to configure protocol specific
```
Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/dsl/BaseIntegrationFlowDefinition.java:2928]:
```suggestion
	 * @param messageHandlerSpec the {@link MessageHandlerSpec} to configure the protocol specific
```

File: src/reference/asciidoc/reactive-streams.adoc [hunk 2]
```diff
@@ -154,6 +154,29 @@ However, when a `ReactiveStreamsConsumer` is involved in the flow (e.g. when cha
 One of the out-of-the-box `ReactiveMessageHandler` implementation is a `ReactiveMongoDbStoringMessageHandler` for Outbound Channel Adapter.
 See <<./mongodb.adoc#mongodb-reactive-channel-adapters,MongoDB Reactive Channel Adapters>> for more information.
 
+Starting with version 6.1, the `IntegrationFlowDefinition` expose a convenient `handleReactive(ReactiveMessageHandler)` terminal operator.
+Any `ReactiveMessageHandler` implementation (even just plain lambda with `Mono` API) can be used for this operator.
+The framework subscribes to the returned `Mono<Void>` automatically.
+Here is a simple sample of possible configuration for this operator:
```
Comment (garyrussell) [inline src/reference/asciidoc/reactive-streams.adoc:160]:
```suggestion
Starting with version 6.1, the `IntegrationFlowDefinition` exposes a convenient `handleReactive(ReactiveMessageHandler)` terminal operator.
Any `ReactiveMessageHandler` implementation (even just a plain lambda using the `Mono` API) can be used for this operator.
The framework subscribes to the returned `Mono<Void>` automatically.
Here is a simple sample of possible configuration for this operator:
```

File: src/reference/asciidoc/reactive-streams.adoc [hunk 3]
```diff
@@ -154,6 +154,29 @@ However, when a `ReactiveStreamsConsumer` is involved in the flow (e.g. when cha
...
+    return IntegrationFlow.from("tappedChannel1")
+            .wireTap("tapChannel", wt -> wt.selector(m -> m.getPayload().equals("foo")))
+            .handleReactive((message) -> Mono.just(message).log().then());
+}
+----
+====
+
+An overloaded version of this operator accepts a `Consumer<GenericEndpointSpec<ReactiveMessageHandlerAdapter>>` to customize a consumer endpoint around the provided `ReactiveMessageHandler`.
+
+In addition, a `ReactiveMessageHandlerSpec`-based variants also provided.
```
Comment (garyrussell) [inline src/reference/asciidoc/reactive-streams.adoc:176]:
```suggestion
In addition, a `ReactiveMessageHandlerSpec`-based variants are also provided.
```

**Review by garyrussell — CHANGES_REQUESTED** [review]
Just doc polishing.

File: spring-integration-core/src/main/java/org/springframework/integration/dsl/BaseIntegrationFlowDefinition.java [hunk 4]
```diff
@@ -2916,6 +2920,68 @@ public IntegrationFlow nullChannel() {
...
+
+		return handleReactive(messageHandlerSpec, null);
+	}
+
+	/**
+	 * Populate a terminal consumer endpoint for the selected protocol specific
+	 * {@link MessageHandler} implementation
+	 * from the respective namespace factory (e.g. {@code Http, Kafka, Files}).
+	 * In addition, accept options for the integration endpoint using {@link GenericEndpointSpec}.
+	 * @param messageHandlerSpec the {@link MessageHandlerSpec} to configure protocol specific
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/dsl/BaseIntegrationFlowDefinition.java:2945]:
```suggestion
	 * @param messageHandlerSpec the {@link MessageHandlerSpec} to configure the protocol specific
```

**Review by garyrussell — APPROVED** [review]

**Pull request merged**